lib/cart.rb

```diff
@@ -0,0 +1,59 @@
+require_relative "menu_item"
+require_relative "menu"
+require_relative "order_item"
+require_relative "customer"
+
+class Cart
+  attr_accessor :cart
+  attr_accessor :total_cost
+
+  def initialize
+    @cart = []
+    @total_cost = 0
+  end
+
+  @@instance = Cart.new
+
+  def self.instance
+    return @@instance
+  end
+
+  def add_item_to_cart(order_item)
+    @cart.push(order_item)
+    cost = order_item.cost.to_i
+    @total_cost += cost
+  end
+
+  def to_s
+    puts "You ordered:"
+    cart.each do |order_item|
+      puts "#{order_item.name}: #{order_item.quantity} x #{order_item.unit_price} #{order_item.currency} = #{order_item.cost} #{order_item.currency}"
+    end
+    puts "GRAND TOTAL: #{@total_cost}"
+  end
+
+  private_class_method :new
+end
+
+# The following was written so I could try out something from
+# https://www.ruby-lang.org/en/documentation/quickstart/
+# which Thomas had us go through on Thursday. What it does
+# "__FILE__ is the magic variable that contains the name of
+# the current file. $0 is the name of the file used to start
+# the program. This check says “If this is the main file
+# being used…” This allows a file to be used as a library,
+# and not to execute code in that context, but if the file is
+# being used as an executable, then execute that code.""
+if __FILE__ == $0
```

Comment:
So, this is a fine technique but in our case I, as we have access to a testing framework, I would argue that this should be in a spec file as a feature test.